calico/datamodel_v1.py

```diff
@@ -220,3 +225,33 @@ def __ne__(self, other):
 
     def __hash__(self):
         return hash(self.endpoint) + hash(self.workload)
+
+
+class TieredPolicyId(object):
+    __slots__ = ["tier", "policy_id"]
+
+    def __init__(self, tier, profile_id):
+        # Intern the strings, which may occur in many IDs.
+        self.tier = intern(tier.encode("utf8"))
+        self.policy_id = intern(profile_id.encode("utf8"))
```

Comment:
What is the encoding for?  Are tier  and profile_id passed in as byte arrays, and then converted to strings?  Or vice versa?